### ISSUE TITLE
Allow blitz jobs to be handled in separate queue

### ISSUE THREAD

**Issue by SteffiFredmansky** [putyourlightson/craft-blitz#577]
We have the problem that due to many pages in our website the blitz cache generation is blocking the queue for quite some time. Can you provide a setting so the blitz jobs can be handled in a separate (custom) queue?

For now we have tried to give the blitz jobs less priority and we have tried to reduce the number of pages that are going to be served by the blitz cache.

**Comment by bencroker** [comment]
When you say it is “blocking” the queue, do you mean that the cache generation queue job is taking a long time? I’m not sure it even is possible to have a custom queue in Craft.

Ideally, you set things up so that your site has to re-generate as few pages as possible, as explained at https://putyourlightson.com/plugins/blitz#best-practices

**Comment by SteffiFredmansky** [comment]
Yes, it is taking really long.

And yes, you can have a custom queue in craft. It is defined in config/app.php, eg.:
```
'customQueue' => [
            'class' => Queue::class,
            'channel' => 'custom',
],
```

It would be great, if we could pass such a custom queue to blitz in the settings.

**Comment by bencroker** [comment]
I’ll need to carefully consider what that might involve. 

In the meantime, how many URLs are being generated, how long is it taking on average and what cache generator driver are you using (HTTP or Local)?

**Comment by bencroker** [comment]
Any update on this, @SteffiFredmansky? And have you tried implementing the best practices as per https://putyourlightson.com/plugins/blitz#best-practices ?

**Comment by bencroker** [comment]
FYI I’m revisiting making the queue configurable in Blitz since talking to @sarahschuetz, and should be able to add it for the next release.

**Comment by bencroker** [comment]
I’ve added the ability to configure a custom queue component in https://github.com/putyourlightson/craft-blitz/commit/ab0c7f85fb23eef8b7439a017c529358d3836827. You’ll be able to do this from `config/app.php`:

```php
return [
    'components' => [
        'plugins' => [
            'pluginConfigs' => [
                'blitz' => [
                    'queue' => 'customQueue',
                ],
            ],
        ],
    ],
];
```

You can test this by running `composer require "putyourlightson/craft-blitz:dev-feature/custom-queue as 4.9.0"`.

**Comment by sarahschuetz** [comment]
Thank you @bencroker! 🎉 Just wanted to share the link as we talkes about only to find that you already implemented it ❤️

**Comment by bencroker** [comment]
Let me know once you’ve had a chance to test, @SteffiFredmansky.

**Comment by mnlmaier** [comment]
@bencroker Hate to be that guy again 🫣 but @sarahschuetz convinced me to ask anyway: We are experiencing the issue in the same Craft 3 project I opened a Campaign issue for last week (https://github.com/putyourlightson/craft-campaign/issues/433). Any chance that this might make it in a Craft v3 compatible release as well? This would be greatly appreciated and would help us with a lot of tasks we're currently working on.

**Comment by bencroker** [comment]
I can revisit once we get the Blitz 4 version out. The issue is that the ability to use a custom queue was only added in Craft 3.4.0, whereas Blitz 3 supports as far back as Craft 3.2.1.

**Comment by bencroker** [comment]
Released in versions [4.9.0](https://github.com/putyourlightson/craft-blitz/releases/tag/4.9.0) and [3.13.0](https://github.com/putyourlightson/craft-blitz/releases/tag/3.13.0).
